File: src/main/java/net/slipp/support/jpa/CreatedAndUpdatedDateEntityListener.java
package net.slipp.support.jpa;

import java.util.Date;

import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;

public class CreatedAndUpdatedDateEntityListener {

	@PrePersist
	public void prePersist(HasCreatedAndUpdatedDate hcud) {
		Date currentDate = getCurrentDate();
		hcud.setCreatedDate(currentDate);
		hcud.setUpdatedDate(currentDate);
	}

	private Date getCurrentDate() {
		return new Date();
	}

	@PreUpdate
	public void preUpdate(HasCreatedAndUpdatedDate hcud) {
		Date currentDate = getCurrentDate();
		hcud.setUpdatedDate(currentDate);
	}
}

File: src/main/scala/net/slipp/support/jpa/CreatedAndUpdatedDateEntityListener.scala
package net.slipp.support.jpa

import java.util.Date
import javax.persistence.PrePersist
import javax.persistence.PreUpdate

class CreatedAndUpdatedDateEntityListener {
  @PrePersist def prePersist(hcud: HasCreatedAndUpdatedDate) {
    val currentDate: Date = getCurrentDate
    hcud.setCreatedDate(currentDate)
    hcud.setUpdatedDate(currentDate)
  }

  private def getCurrentDate: Date = {
    return new Date
  }

  @PreUpdate def preUpdate(hcud: HasCreatedAndUpdatedDate) {
    val currentDate: Date = getCurrentDate
    hcud.setUpdatedDate(currentDate)
  }
}
